### ISSUE TITLE
Add dave and bambi mechanics

### ISSUE THREAD

**Issue by PHGaming121** [Theoyeah/Theoyeah-Engine#233]
### What feature do you want to get added on the **base** engine?

Add Dave and Bambi events

List of events i want:
- [ ] Eyesores
   - What are eyesores?
- [x] Press 7 do something
   - Value 1: thing you want to it to do
   - Value 2: duration of this event
   - Value 3: specifications of the specified thing (Value 1)
- [x] Window move mechanic
   - Value 1: x
   - Value 2: y
- [x] Window pop-up event
   - Value 1: title of the alert
   - Value 2: text
- [ ] Wavy effect on Background
   - I don’t freaking know how to do this, but I think you can do it with lua, and also with the _Add Shader_ event

Wither talks: ok, I’m will going to create this in 15 days (aprox), meanwhile, I will specify the things that each event should have, and I’ll also add more events to this list, @Theoyeah, please, don’t close this

- [ ] fix null object reference when opening the game (mark this when fixed plz)

**Comment by Wither362** [comment]
Before anything else, @PHGaming121, ¿are you doing a Dave and Bambi mod?

**Comment by Theoyeah** [comment]
> Before anything else, @PHGaming121, are you doing a Dave and Bambi mod?

You might want to create a branch for is ideas, as i don't feel like adding it directly to the main branch

**Comment by Theoyeah** [comment]
> > Before anything else, @PHGaming121, are you doing a Dave and Bambi mod?
> 
> You might want to create a branch for his ideas, as i don't feel like adding it directly to the main branch

**Comment by PHGaming121** [comment]
1st: eyesores are an type of an screenshader that u see in VS Strident Crisis mod (the colorful thing that appears in screen in some songs)

2nd i think the press 7 thing look like this
![Captura de Tela (192)](https://user-images.githubusercontent.com/76562427/184383892-689a17ab-0ce0-4e44-9ee2-f2bbb3ed913a.png)
case of load song: type the song you want
case play video; type the name of the video u want
(textbox needed for this)

3rd window move mechanic looks like value 1 = x pos, value 2 = y pos

4th the window popup is exactly what you're doing

5th HOW DO I USE ADD SHADER??? (for wavy bg lol)

well... this is what i can do

**Comment by PHGaming121** [comment]
and yes im doing a "private" D&B mod

**Comment by PHGaming121** [comment]
basically, i want someone good at coding for porting the main mechanics of the Strident Crisis mod to PE 0.6.2 (want PE 0.6.2 because the scripts folder)
If u can, plz add some of your engine to this port

**Comment by PHGaming121** [comment]
more things u should add

list of events
![2](https://user-images.githubusercontent.com/76562427/184411597-27569813-33e8-416f-9394-8aa087a432ee.jpg)

press 7 event (i already send it lol)
![Captura de Tela (192)](https://user-images.githubusercontent.com/76562427/184411896-b02942c4-c1c9-4820-9d63-5c9071fcb0e7.png)

Different noteskin for opponent and Boyfriend
![Captura de Tela (193)](https://user-images.githubusercontent.com/76562427/184412131-8defef78-f8fb-4515-a7cb-8de5c7ab96ff.png)

No Ghost tapping(i know i can set it with the setProperty event), and a watemark text

![Captura de Tela (194)](https://user-images.githubusercontent.com/76562427/184412756-9adac981-b166-4c52-bcf2-0a11c618ca4f.png)
![Captura de Tela (195)](https://user-images.githubusercontent.com/76562427/184412759-1b31c324-2adb-484a-a300-df5c60fba9e9.png)

**Comment by Wither362** [comment]
> From @PHGaming121:
>
> more things u should add
> 
> list of events
![2](https://user-images.githubusercontent.com/76562427/184411597-27569813-33e8-416f-9394-8aa087a432ee.jpg)
> 
> press 7 event (i already send it lol)
![Captura de Tela (192)](https://user-images.githubusercontent.com/76562427/184411896-b02942c4-c1c9-4820-9d63-5c9071fcb0e7.png)
> 
> Different noteskin for opponent and Boyfriend
![Captura de Tela (193)](https://user-images.githubusercontent.com/76562427/184412131-8defef78-f8fb-4515-a7cb-8de5c7ab96ff.png)
> 
> No Ghost tapping (i know i can set it with the setProperty event), and a watemark text
> 
> ![Captura de Tela (194)](https://user-images.githubusercontent.com/76562427/184412756-9adac981-b166-4c52-bcf2-0a11c618ca4f.png)
![Captura de Tela (195)](https://user-images.githubusercontent.com/76562427/184412759-1b31c324-2adb-484a-a300-df5c60fba9e9.png)

Oh my fucking god, do I have to do that? Well, it will take some time<br><br>

> From @Theoyeah:
>
> > > Before anything else, @PHGaming121, are you doing a Dave and Bambi mod?
> 
> You might want to create a branch for his ideas, as i don't feel like adding it directly to the main branch

What do you mean with that?

**Comment by Theoyeah** [comment]
That we should create a different branch for it

**Comment by Wither362** [comment]
> That we should create a different branch for it

Yeah, but if we Crete a new branch, we’ll have to be refreshing it with the latest changes every time we commit, so why don’t we create a branch, and than merge it to the main one?

**Comment by Theoyeah** [comment]
That's easy to update the branch, just a click and were done. Atf i can do it

**Comment by Wither362** [comment]
ok, @PHGaming121, may i join to your mod please? i love dave and bambi

**Comment by Wither362** [comment]
Im with it right now!

**Comment by Wither362** [comment]
here you have it: https://github.com/Theoyeah/Theoyeah-Engine/tree/10--For-dave-and-bambi-lovers

**Comment by PHGaming121** [comment]
it saying null object reference when i open the engine

**Comment by PHGaming121** [comment]
@Wither362 how to fix this

**Comment by Wither362** [comment]
> @Wither362 how to fix this

Let me play it later and I will fix it, btw, give me a photo of the error it gives please

**Comment by PHGaming121** [comment]
> > @Wither362 how to fix this
> 
> Let me play it later and I will fix it, btw, give me a photo of the error it gives please

here:
![error](https://user-images.githubusercontent.com/76562427/186904197-fff028b5-b57e-4ea9-a109-2364f582aac0.png)

**Comment by Wither362** [comment]
> 

wtf

**Comment by PHGaming121** [comment]
sooooo.... i see an new update on the code but..... same thing, it doesnt even open the game, just the saaaaame popup

**Comment by Wither362** [comment]
> 

idk what is happening and i need help with it
